### PR TITLE
added app insight telemetry-phase1 for azure jenkins storage plugin

### PR DESCRIPTION
**Pull request by arroyc**

### CODE REVIEW

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java [hunk 1]
```diff
@@ -271,562 +268,566 @@ private static String getCustomURI(String storageAccountName, String type,
...
+                // blob reference.
+                String blobURI = zipPath.getName();
+
+                if (!Utils.isNullOrEmpty(expVP)) {
+                    blobURI = expVP + blobURI;
+                }
+
+                CloudBlockBlob blob = container.getBlockBlobReference(blobURI);
+
+                telemetry.createEvent("AzureJenkinsStorageUploadEvent", filesUploaded + " files uploaded, upload type " + uploadType.toString());
```
Comment (clguiman) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java:432]: shouldn't we log the file sizes?

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureBlobAction.java [hunk 2]
```diff
@@ -138,15 +143,19 @@ public void doProcessDownloadRequest(final StaplerRequest request, final Stapler
 		for (AzureBlob blob : individualBlobs) {
 			if (blob.getBlobName().equals(blobName)) {
 				try {
-					response.sendRedirect2(blob.getBlobURL()+"?"+getSASURL(accountInfo));
+                                    telemetry.createEvent(telemetryEvent, accountInfo.getStorageAccName()+" downloaded blob "+ blob.getBlobName());
```
Comment (clguiman) [inline src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureBlobAction.java]: we shouldn't log blob names

Comment (arroyc) [reply src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureBlobAction.java]: agreed, we should not actually overload our telemetry with such, I'm planning to remove this event. No need to track each and every blob, we just care about total files and may be total size

File: src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java (hunk 1 above)
Comment (arroyc) [reply src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java:432]: right now there is no implementation in the code  that keep tracks of blob size, we can implement it. will wait for Arun ...

**Pull request closed without merging**